code/io/tree_binary.c

```diff
@@ -20,13 +20,15 @@ FILE *load_fd;
 
 // External Functions //
 
+#define MAX_BUF_SIZE (3*MAX_STRING_LEN+40)
+
 void load_tree_table_binary(int32_t filenr) 
 {
   int i, totNHalos;
-  char buf[MAX_STRING_LEN];
+  char buf[MAX_BUF_SIZE];
```

Comment:
Perhaps allocate to `MAX_BUF_SIZE + 1` (to avoid passing `MAX_BUF_SIZE - 1` to `snprintf`). This would be true here and elsewhere in this PR

Reply:
I have no particular preference either way, though I would tend to agree. I guess we could ask "which convention is a typical cosmologist most likely to expect?"

Reply:
OK, I see that the simplest option is for me to propose an updated pull request. And that the `- 1` only occurs in one place. I'm preparing an update...

Reply:
Yeah the best way is to pass `MAX_STRING_LEN - 1` as the second parameter to `snprintf`. That way calculations for `MAX_BUF_SIZE` will still be correct.